config.yaml

```diff
@@ -1,2 +1,40 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
+
+options:
+  client_id:
+    description: The registered client_id
+    type: string
+  client_secret:
+    description: The registered client_secret
+    type: string
+  provider:
+    default: generic
+    description: |
+      The provider name, must be one of the following:
+      ["generic", "google", "facebook", "microsoft", "github",
+      "apple", "gitlab", "auth0", "slack", "spotify", "discord",
+      "twitch", "netid", "yandex", "vk", "dingtalk"].
+      Defaults to "generic"
+    type: string
+  issuer_url:
+    description: |
+      The issuer_url, this value is only used when provider is "generic" or "auth0"
+      type: string
+    type: string
+  secret_backend:
+    description: |
+      The backend to use for passing sensitive information to Kratos
+    type: string
+  microsoft_tenant_id:
```

Comment:
This is out of scope for this PR, but do we plan to support multitenancy?

Reply:
Multitenancy as in hosting the credentials for multiple providers in 1 integrator unit? Probably not, this would make the config harder to manage.